src/Service/Query/SynonymRule.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace SilverStripe\Forager\Service\Query;
+
+use SilverStripe\Core\Injector\Injectable;
+
+class SynonymRule
```

Comment:
this and the related items could use a json serialise function as currently running it through `json_encode` returns an empty object

Reply:
Thanks, @blueo!

As discussed on Slack, sounds like this needs `JsonSerializable`. I'll add some tests as well.

Reply:
@blueo rebased with `main`, and `JsonSerializable` added here:
https://github.com/silverstripeltd/silverstripe-forager/pull/20/commits/b62fdc95c42dc8c6e5a5a31ec92bbe925a758d05

Reply:
working well - some linting failures have popped up FYI :)